From 459c5c1bd7a6ab4b9598d2ecf885f0ec379d5c78 Mon Sep 17 00:00:00 2001
From: Stanley Chu <yschu@nuvoton.com>
Date: Fri, 29 Jul 2022 15:57:54 +0800
Subject: [PATCH] set clock divisor for uart456

Signed-off-by: Stanley Chu <yschu@nuvoton.com>
---
 board/nuvoton/arbel/arbel.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/board/nuvoton/arbel/arbel.c b/board/nuvoton/arbel/arbel.c
index c5f830b326..a0f5abd3ec 100644
--- a/board/nuvoton/arbel/arbel.c
+++ b/board/nuvoton/arbel/arbel.c
@@ -26,8 +26,10 @@
 DECLARE_GLOBAL_DATA_PTR;
 
 #define CLKSEL	0x4
+#define CLKDIV3    0x58
 #define PIXCKSEL_GFX	0
 #define PIXCKSEL_MASK	GENMASK(5, 4)
+#define CLKDIV3_UARTDIV2_MASK  GENMASK(15, 11)
 #define SR_MII_CTRL_SWR_BIT15	15
 
 #define DRAM_512MB_ECC_SIZE	0x1C000000ULL
@@ -102,6 +104,12 @@ static void arbel_clk_init(void)
 	val &= ~PIXCKSEL_MASK;
 	val |= FIELD_PREP(PIXCKSEL_MASK, PIXCKSEL_GFX);
 	writel(val, NPCM_CLK_BA + CLKSEL);
+
+	/* Set clock divisor 19 for uart4/5/6 */
+	val = readl(NPCM_CLK_BA + CLKDIV3);
+	val &= ~CLKDIV3_UARTDIV2_MASK;
+	val |= FIELD_PREP(CLKDIV3_UARTDIV2_MASK, 19);
+	writel(val, NPCM_CLK_BA + CLKDIV3);
 }
 
 int board_init(void)
-- 
2.34.1

