From fbd1a65d11a9c2be054f7b3661c9cce68e75c15a Mon Sep 17 00:00:00 2001
From: Joseph Liu <kwliu@nuvoton.com>
Date: Mon, 27 Jun 2022 16:56:22 +0800
Subject: [PATCH] rtc nct3018y: enable intruder

Signed-off-by: Joseph Liu <kwliu@nuvoton.com>
---
 drivers/rtc/rtc-nct3018y.c | 18 +++++++++++++++++-
 1 file changed, 17 insertions(+), 1 deletion(-)

diff --git a/drivers/rtc/rtc-nct3018y.c b/drivers/rtc/rtc-nct3018y.c
index 3d018526948d..94773d9effcd 100644
--- a/drivers/rtc/rtc-nct3018y.c
+++ b/drivers/rtc/rtc-nct3018y.c
@@ -23,6 +23,7 @@
 #define NCT3018Y_REG_CTRL	0x0A /* timer control */
 #define NCT3018Y_REG_ST		0x0B /* status */
 #define NCT3018Y_REG_CLKO	0x0C /* clock out */
+#define NCT3018Y_REG_INST	0x12 /* Intruder Control */
 
 #define NCT3018Y_BIT_AF		BIT(7)
 #define NCT3018Y_BIT_ST		BIT(7)
@@ -38,6 +39,13 @@
 #define NCT3018Y_REG_CLKO_F_MASK	0x03 /* frequenc mask */
 #define NCT3018Y_REG_CLKO_CKE		0x80 /* clock out enabled */
 
+#define NCT3018Y_REG_INST_IRSTEN3	BIT(7)
+#define NCT3018Y_REG_INST_IRSTEN2	BIT(6)
+#define NCT3018Y_REG_INST_IRSTEN1	BIT(5)
+#define NCT3018Y_REG_INST_IRSTEN0	BIT(4)
+#define NCT3018Y_REG_INST_INTRF		BIT(1)
+#define NCT3018Y_REG_INST_INTRIE	BIT(0)
+
 struct nct3018y {
 	struct rtc_device *rtc;
 	struct i2c_client *client;
@@ -495,7 +503,15 @@ static int nct3018y_probe(struct i2c_client *client,
 	flags = 0;
 	err = i2c_smbus_write_byte_data(client, NCT3018Y_REG_ST, flags);
 	if (err < 0) {
-		dev_err(&client->dev, "%s: write error\n", __func__);
+		dev_err(&client->dev, "%s: Unable to clear status\n", __func__);
+		return err;
+	}
+
+
+	flags = NCT3018Y_REG_INST_INTRIE | NCT3018Y_REG_INST_INTRF;
+	err = i2c_smbus_write_byte_data(client, NCT3018Y_REG_INST, flags);
+	if (err < 0) {
+		dev_err(&client->dev, "%s: Unable enable intruder\n", __func__);
 		return err;
 	}
 
-- 
2.34.1

